### PR TITLE
Added Doctor

### PR DESCRIPTION
**Pull request by squidserv**
Added Doctor convars
Moved Assassin to the right section in sv_terrortown.lua

### CODE REVIEW

File: lua/ulx/xgui/gamemodes/terrortown.lua [hunk 1]
```diff
@@ -294,6 +294,15 @@ local function AddSpecialistInnocentSettings(gppnl)
 
     local minvet = xlib.makeslider { label = "ttt_veteran_min_players (def. 0)", min = 0, max = 10, repconvar = "rep_ttt_veteran_min_players", parent = spinnlst }
     spinnlst:AddItem(minvet)
+
+    local hasdoc = xlib.makecheckbox { label = "ttt_Doctor_enabled (def. 0)", repconvar = "rep_ttt_doctor_enabled", parent = spinnlst }
```
Comment (Malivil) [inline lua/ulx/xgui/gamemodes/terrortown.lua:298]: Minor: the 'D' in Doctor should be lowercase here

**Review by Malivil — CHANGES_REQUESTED** [review]
All SHOP_ROLES have `ttt_shop_random_[role_string_short]_percent` and `ttt_shop_random_[role_string_short]_enabled` automatically added so you should add those to the ULX as well

File: lua/ulx/xgui/gamemodes/terrortown.lua (hunk 1 above)
Comment (squidserv) [reply lua/ulx/xgui/gamemodes/terrortown.lua:298]: Oof good catch

**Comment by squidserv** [comment]
> All SHOP_ROLES have `ttt_shop_random_[role_string_short]_percent` and `ttt_shop_random_[role_string_short]_enabled` automatically added so you should add those to the ULX as well

The Doctor isn't a SHOP_ROLE yet

**Comment by Malivil** [comment]
Ah, OK. Something to keep in mind for later then =)

**Review by Malivil — APPROVED** [review]

**Pull request merged**